assets/src/js/Components/ArchivePicker.js

```diff
@@ -1,151 +1,115 @@
-        pagedTitle: `${ pageIndex } -- ${ image.title }`
-      } ) );
-      this.setState( {
-        currentPage: pageIndex,
-        images: [
-          ...( newSearch ? [] : this.state.images ),
-          ...withPageLabel
-        ],
-        searchText: searchedText,
-      } );
-    } catch ( error ) {
-      this.setState( { error } );
-    } finally {
-      this.setState( { loading: false } );
-    }
-  }
+  const toggleMultiSelection = target => setSelectedIds(
+    selectedIds.includes(target.id)
+      ? selectedIds.filter(id => id !== target.id)
+      : [...selectedIds, target.id]
+  );
 
-  async includeInWp( ids ) {
-    try {
-      this.setState( { processingImages: true } );
-      const updatedImages = await apiFetch( {
-        method: 'POST',
-        path: '/planet4/v1/image-archive/transfer',
-        data: {
-          ids: ids,
-          use_original_language: false,
-        }
-      } );
-      this.updateFromUploadedResponse( updatedImages );
-    } catch ( e ) {
-      console.log( e );
-      this.setState( { processingError: e } );
-    } finally {
-      this.setState( { processingImages: false } );
-    }
-  }
-
-  updateFromUploadedResponse( updatedImages ) {
-    const newImages = this.state.images.map( stateImage => {
-      const updated = updatedImages.find( updatedImage => updatedImage.id === stateImage.id );
-      if ( updated ) {
-        return updated;
-      }
-      return stateImage;
-    } );
-    this.setState( {
-      images: newImages,
-    } );
-  }
-
-  async search() {
-    await this.loadNextPage( true );
-  }
+  useEffect(() => {
+    loadPage(currentPage, searchText);
+  }, [currentPage, searchText]);
 
-  render() {
-    const {
-      loading,
-      error,
-      images,
-    } = this.state;
+  const selectedImages = getSelectedImages(selectedIds);
 
-    return <Fragment>
-      <form
-        className={ 'archive-picker-search' }
-        onSubmit={ event => {
-          event.preventDefault();
-          this.search();
-        } }
-        onChange={ event => this.setState( { enteredSearch: event.target.value } ) }
+  return <Fragment>
+    <form
+      className={'archive-picker-search'}
+      onSubmit={event => {
+        event.preventDefault();
+        if (!loading) {
+          setSearchText(enteredSearch);
+          setCurrentPage(1);
+        }
+      }}
+      onChange={event => setEnteredSearch(event.target.value)}
+    >
+      <input
+        type='text'
+        disabled={loading}
+      />
+      <input type='submit' value={__('Search', 'planet4-master-theme-backend')} />
+    </form>
+    {loading && (
+      <div className={'archive-picker-loading'}> loading...</div>
+    )}
+    {!!error && (
+      <div>
+        <h3>API error:</h3>
+        <p> {error.message} </p>
+      </div>
+    )}
+    <div className={'image-picker'}>
+      <ul
+        className={'picker-list'}
+        onScroll={event => {
+          if (isNearScrollEnd(event) && !loading) {
```

Comment:
Can you swap these so that it doesn't need to calculate the position if we're scrolling while loading?